pkg/endpoint/endpoint.go

```diff
@@ -2076,6 +2077,21 @@ func (e *Endpoint) Delete(monitor monitorOwner, ipam ipReleaser, manager endpoin
 		}
 	}
 
+	if option.Config.IPAM == option.IPAMENI {
```

Comment:
I skimmed through the PR but did not do a thorough review. 1st questions why not do the delete at the CNI plugin level rather then in the agent at endpoint deletion time?

Reply:
The reason is mostly convenience. We know that all the deletion logic will eventually flow through the endpoint level. There was a bug recently about cilium-health not working in ENI mode and that was because it's an endpoint, but its IPs are not assigned by the CNI. We fixed that in https://github.com/cilium/cilium/pull/11073. 

Not all things in Cilium are allocated at the CNI level, so it made sense to keep things in one place and that was the endpoint level. 

However, there isn't a strict rule. We can totally move this to the CNI level and also add this deletion logic for the cilium-health endpoint as well. I'm not tied to any one way, but I did it at the endpoint level just so I didn't have to add the same code to two different spots.

Reply:
Ok cool, I see your point of view as well here. Maybe this should be tackled if we ever decide to refactor how cilium-health works.

Would be nice to have the guarantee that there is `one` component doing the creation and cleanup of all things on the host netns related to IPAM configs.